core/src/raw/oio/buf/queue_buf.rs

```diff
@@ -78,6 +78,27 @@ impl QueueBuf {
         }
     }
 
+    /// Convert to a new [`Buffer`] from the queue.
+    ///
+    /// If the queue is empty, it will return an empty buffer. Otherwise, it will iterate over all
+    /// buffers and collect them into a new buffer.
+    ///
+    /// # Notes
+    ///
+    /// There are allocation overheads when collecting multiple buffers into a new buffer. But
+    /// most of them should be acceptable since we can expect the item length of buffers are slower
+    /// than 4k.
+    #[inline]
+    pub fn into_buffer(mut self) -> Buffer {
```

Comment:
Seems a good idea. How about changing `collect` into `fn collect(mut self) -> Buffer` instead? This change allow align with `Iterator`'s `collect` semantic.

Reply:
> Seems a good idea. How about changing `collect` into `fn collect(mut self) -> Buffer` instead?

I'm willing to do this change if you also like this idea.

Reply:
Updated.